Rocket.Chat/Controllers/Auth/ConnectServerViewController.swift

```diff
@@ -78,7 +100,25 @@ final class ConnectServerViewController: BaseViewController {
             textFieldServerURL.text = applicationServerURL.host
             connect()
         }
+        NotificationCenter.default.addObserver(self, selector: #selector(internetConnection), name: Notification.Name.reachabilityChanged, object: reachability)
+                do {
+                        try reachability.startNotifier()
+                    } catch {
+                            print(error)
+                        }
+    }
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewDidDisappear(true)
+                    NotificationCenter.default.removeObserver(self)
```

Comment:
I guess it'll be better if you use removeObserver in deinit method. The page may call addObserve method in viewDidLoad, for example, and after viewWillDisappear it will removed from NotificationCenter. It's incorrect logic.